src/Algebra/Construct/Sub/Ring/Ideal.agda

```diff
@@ -0,0 +1,26 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Ideals of a ring
+------------------------------------------------------------------------
+
+{-# OPTIONS --safe --cubical-compatible #-}
+
+open import Algebra.Bundles using (Ring)
+
+module Algebra.Construct.Sub.Ring.Ideal {c ℓ} (R : Ring c ℓ) where
+
+open import Algebra.Module.Construct.Sub.Bimodule using (Subbimodule)
+open import Algebra.Module.Construct.TensorUnit using (bimodule)
+open import Level
+
+------------------------------------------------------------------------
+-- Definition
+-- a (two sided) ideal is exactly a subbimodule of R considered as a bimodule over itself
```

Comment:
minor: sub-bimodule please. I find 'subbimodule' very hard to parse (as a human) properly.